### PR TITLE
[SPARK-24411] [SQL] Adding native Java tests for 'isInCollection'

### PR DESCRIPTION
**Pull request by aai95**
## What changes were proposed in this pull request?
`JavaColumnExpressionSuite.java` was added and `org.apache.spark.sql.ColumnExpressionSuite#test("isInCollection: Java Collection")` was removed.
It provides native Java tests for the method `org.apache.spark.sql.Column#isInCollection`.

### CODE REVIEW

**Comment by aai95** [comment]
Sorry for misunderstanding, @dbtsai do you mean that the goal of this issue is in creation of a new Java test class with one test which will replace current Scala test "isInCollection: Java Collection"?
Or the goal is in rewriting all tests from Scala class "ColumnExpressionSuite" to Java code?

**Comment by dbtsai** [comment]
Historically, we test many Java APIs in Scala suite, and hope it will work in Java. But that fails us sometime. The goal is to add `JavaColumnExpressionSuite.java`, and move `test("isInCollection: Java Collection")` to Java suite.

**Comment by aai95** [comment]
@dbtsai review my new changes please.
Should I delete `test("isInCollection: Java Collection")` from `ColumnExpressionSuite.scala` or make changes in `JavaColumnExpressionSuite.java`?

**Comment by dbtsai** [comment]
ok to test

**Comment by dbtsai** [comment]
Remove `test("isInCollection: Java Collection")` from the scala code, and please follow the code style in other java code. Remember to add the ASF Source Header.

File: sql/core/src/test/java/test/org/apache/spark/sql/JavaColumnExpressionSuite.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
+package test.org.apache.spark.sql;
```
Comment (viirya) [inline sql/core/src/test/java/test/org/apache/spark/sql/JavaColumnExpressionSuite.java]: You need to add the Apache License header at the beginning of a file.

File: sql/core/src/test/java/test/org/apache/spark/sql/JavaColumnExpressionSuite.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.*;
+
+import static org.apache.spark.sql.types.DataTypes.*;
+
+public class JavaColumnExpressionSuite {
+
+    private transient TestSparkSession spark;
```
Comment (HyukjinKwon) [inline sql/core/src/test/java/test/org/apache/spark/sql/JavaColumnExpressionSuite.java]: Java file also should be two space indentation

File: sql/core/src/test/java/test/org/apache/spark/sql/JavaColumnExpressionSuite.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import org.apache.spark.sql.test.TestSparkSession;
+import org.apache.spark.sql.types.StructType;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.*;
+
+import static org.apache.spark.sql.types.DataTypes.*;
```
Comment (HyukjinKwon) [inline sql/core/src/test/java/test/org/apache/spark/sql/JavaColumnExpressionSuite.java]: Shall we avoid wlidcard import if they are less then 5 imports

Comment (aai95) [reply sql/core/src/test/java/test/org/apache/spark/sql/JavaColumnExpressionSuite.java]: Hello @HyukjinKwon, I tried to remove the wildcard import and got the following result:

`import java.util.ArrayList;`
`import java.util.Arrays;`
`import java.util.HashSet;`
`import java.util.List;`
`import java.util.Locale;`

`import static org.apache.spark.sql.types.DataTypes.IntegerType;`
`import static org.apache.spark.sql.types.DataTypes.StringType;`
`import static org.apache.spark.sql.types.DataTypes.createArrayType;`
`import static org.apache.spark.sql.types.DataTypes.createStructField;`
`import static org.apache.spark.sql.types.DataTypes.createStructType;`

Is it OK to use wildcard here if exactly 5 imports were used in every case? Thanks.

**Comment by aai95** [comment]
I took into account all your remarks. Thanks.
@dbtsai @viirya @HyukjinKwon review my changes please.
1. Scala test was removed.
2. ASF License Header was added.
3. The name of the 2nd method was shortened.
4. Changes were made in the Java code style.
5. A bug was fixed if the method did not throw an exception.

**Comment by dbtsai** [comment]
LGTM. Merged into master. Thanks.

**Pull request closed without merging**